### PR TITLE
fix: Issues related to Ledger variant

### PR DESCRIPTION
**Pull request by the-prateek-mishra**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.ts [hunk 1]
```diff
@@ -1884,8 +1886,12 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
      * @memberof NewLedgerEntryPanelComponent
      */
     private loadStockVariants(stockUniqueName: string): void {
-        this.stockVariants$ = this.ledgerService.loadStockVariants(stockUniqueName).pipe(
-            map((variants: IVariant[]) => variants.map((variant: IVariant) => ({label: variant.name, value: variant.uniqueName}))));
+        this.ledgerService.loadStockVariants(stockUniqueName).pipe(
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.ts:1889]: add takeUntil

File: apps/web-giddh/src/app/ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.ts [hunk 2]
```diff
@@ -1884,8 +1886,12 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
...
+            map((variants: IVariant[]) => variants.map((variant: IVariant) => ({label: variant.name, value: variant.uniqueName})))).subscribe(res => {
+                this.stockVariants$ = observableOf(res);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.ts:1891]: check if error response is handled

File: apps/web-giddh/src/app/ledger/components/update-ledger-entry-panel/update-ledger-entry-panel.component.ts [hunk 3]
```diff
@@ -2552,8 +2554,12 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
      * @memberof UpdateLedgerEntryPanelComponent
      */
     private loadStockVariants(stockUniqueName: string): void {
-        this.stockVariants$ = this.ledgerService.loadStockVariants(stockUniqueName).pipe(
-            map((variants: IVariant[]) => variants.map((variant: IVariant) => ({label: variant.name, value: variant.uniqueName}))));
+        this.ledgerService.loadStockVariants(stockUniqueName).pipe(
+            map((variants: IVariant[]) => variants.map((variant: IVariant) => ({label: variant.name, value: variant.uniqueName})))).subscribe(res => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/update-ledger-entry-panel/update-ledger-entry-panel.component.ts]: add takeUntil and check if error is handled

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.ts (hunk 2 above)
Comment (the-prateek-mishra) [reply apps/web-giddh/src/app/ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.ts:1891]: Done

File: apps/web-giddh/src/app/ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.ts (hunk 1 above)
Comment (the-prateek-mishra) [reply apps/web-giddh/src/app/ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.ts:1889]: Done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**